SonarQube.TeamBuild.PreProcessor/Roslyn/EmbeddedAnalyzerInstaller.cs

```diff
@@ -0,0 +1,208 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="EmbeddedAnalyzerInstaller.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+
+namespace SonarQube.TeamBuild.PreProcessor.Roslyn
+{
+    /// <summary>
+    /// Handles fetching embedded resources from SonarQube plugins
+    /// </summary>
+    /// <remarks>
+    /// We won't be able to run the analyzers unless the user is using MSBuild 14.0 or later.
+    /// However, this code is called during the pre-process stage i.e. we don't know which
+    /// version of MSBuild will be used so we have to download the analyzers even if we
+    /// can't then use them.
+    /// </remarks>
+    public class EmbeddedAnalyzerInstaller //: IAnalyzerInstaller
```

Comment:
This class doesn't implement _IAnalyzerInstaller_ yet as the interface expects a list of NuGet packages rather than a list of plugins. This is changed in a later commit when the NuGet implementation is removed.